pkg/kubelet/cri/remote/remote_image.go

```diff
@@ -98,6 +98,8 @@ func (r *remoteImageService) validateServiceConnection(ctx context.Context, conn
 
 	} else if status.Code(err) == codes.Unimplemented {
 		return fmt.Errorf("CRI v1 image API is not implemented for endpoint %q: %w", endpoint, err)
+	} else if status.Code(err) == codes.Unavailable {
```

Comment:
@saschagrunert Thank you for your PR!

Can you explain why do we check only for 2 codes? There are more than 2 defined in the grpc package.
And why this function returns nil on any errors except of these 2?

Reply:
That's a very good point! We now only have one CRI API to test against, means we can check for any error during connection setup.